compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -596,13 +596,28 @@ class Typer extends Namer
         .withSpan(tree.span)
         .computeNullable()
 
-    def typeSelectOnType(qual: untpd.Tree)(using Context) =
-      typedSelect(untpd.cpy.Select(tree)(qual, tree.name.toTypeName), pt)
+    def javaSelectOnType(qual: Tree)(using Context) =
+      // semantic name conversion for `O$` in java code
+      if !qual.symbol.is(JavaDefined)
+      then
```

Comment:
we usually put the `then` on the same line as the `if`

Reply:
Indeed, that looks better